okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -269,80 +270,113 @@ public void updateModuleCommit(Tenant t,
-      Handler<ExtendedAsyncResult<String>> fut) {
-
-    moduleManager.enableAndDisableCheck(tenant, mdFrom, mdTo, cres -> {
-      if (cres.failed()) {
-        pc.debug("enableAndDisableModule: depcheck fail: " + cres.cause().getMessage());
-        fut.handle(new Failure<>(cres.getType(), cres.cause()));
+  private Future<Void> enableAndDisableModuleFut(String tenantId, TenantInstallOptions options,
+                                                 String moduleFrom, TenantModuleDescriptor td,
+                                                 ProxyContext pc) {
+    Promise<Void> promise = Promise.promise();
+    enableAndDisableModule(tenantId, options, moduleFrom, td, pc, res -> {
+      if (res.failed()) {
+        promise.fail(res.cause());
       } else {
-        pc.debug("enableAndDisableModule: depcheck ok");
-        ead1TenantInterface(tenant, tenantParameters, mdFrom, mdTo, false, pc, res -> {
-          if (res.failed()) {
-            fut.handle(new Failure<>(res.getType(), res.cause()));
+        promise.complete();
+      }
+    });
+    return promise.future();
+  }
+
+  Future<Void> disableModules(String tenantId, TenantInstallOptions options, ProxyContext pc) {
+    Promise<Void> promise = Promise.promise();
+    options.setDepCheck(false);
+    listModules(tenantId, res -> {
+      if (res.failed()) {
+        promise.fail(res.cause());
+        return;
+      }
+      Future<Void> future = Future.succeededFuture();
+      for (ModuleDescriptor md : res.result()) {
+        future = future.compose(x -> enableAndDisableModuleFut(tenantId, options,
+          md.getId(), null, pc));
+      }
+      future.setHandler(promise::handle);
+    });
+    return promise.future();
+  }
+
+  private void enableAndDisableModule2(Tenant tenant, TenantInstallOptions options,
+                                      String moduleFrom, ModuleDescriptor mdTo, ProxyContext pc,
+                                      Handler<ExtendedAsyncResult<String>> fut) {
+
+    moduleManager.get(moduleFrom, resFrom -> {
+      if (resFrom.failed()) {
+        fut.handle(new Failure<>(resFrom.getType(), resFrom.cause()));
+        return;
+      }
+      ModuleDescriptor mdFrom = resFrom.result();
+      if (options.getDepCheck()) {
+        moduleManager.enableAndDisableCheck(tenant, mdFrom, mdTo, cres -> {
+          if (cres.failed()) {
+            pc.debug("enableAndDisableModule: depcheck fail: " + cres.cause().getMessage());
+            fut.handle(new Failure<>(cres.getType(), cres.cause()));
             return;
           }
-          ead5commit(tenant, mdFrom, mdTo, pc, res1 -> {
-            if (res1.failed()) {
-              fut.handle(new Failure<>(ErrorType.USER, res.cause()));
-              return;
-            }
-            fut.handle(new Success<>(mdTo != null ? mdTo.getId() : ""));
-          });
+          enableAndDisableModule2(tenant, options, mdFrom, mdTo, pc, fut);
```

Comment:
This line can probably be replaced by Line 356 if we remove the else statement.

Reply:
I don't follow where that is.. The 353 lines .. have else but a lambda.. So I can't fold things together there..